src/client.ts

```diff
@@ -94,24 +94,27 @@ export class RestClientRequest<ReturnType = any> {
                 clearTimeout(abortTimeout);
             }
         }
+        return response;
+    }
 
-        if (result.status === 404) {
+    public async parse(response: Response): Promise<ReturnType | null> {
```

Comment:
Ah - not quite what I meant. I think this wil feel a bit awkward to use? 
```typescript
const request:Request = someApi.doRequest();
const response:Response = await request.call();
const data:ReturnType = request.parse(response);
```

Why not simply something like: 
```typescript
const response:Response = await someApi.someRequest().response();
const data:ReturnType = someApi.someRequest().call();
```

My point is - it feels a bit off that I have to find the utility method on the request for parsing the response data out. 

If not the above I'd consider wrapping the response itself in a helper class where you can give access to data : 


```typescript
const response:RESTResponse = await someApi.someRequest().response();
const data:ReturnType = response.data();
```
RESTResponse would then give access to the details you need:
response.statusCode 
response.headers
etc.

Reply:
Yea I prefer this. I've just pushed a commit that introduces a `RestResponse` class. 

```ts
const request: RestRequest = someApi.someRequest()
const response: RestResponse = await request.call()
const data: ReturnType | null = await response.data()
```

The response object can be extracted from the RestResponse object

```ts
const request: RestRequest = someApi.someRequest()
const response: Response = (await request.call()).response
```

Reply:
I also moved the `AbortController` up to be a class member so the caller can abort calls (it can still abort itself it the timeout is reached)